### ISSUE TITLE
Use of Faker and stuff

### ISSUE THREAD

**Issue by dbraga** [digitalprog/ruby_provisioning_api#35]
I saw you used the Faker gem in the .all of User.
I'm thinking it's a good idea to uniform it and i should use it also in group.
Do you think we should uniform also this behaviour ?

  FAKE_GROUP_ID = "foo_id"
  FAKE_GROUP_NAME = "foo_name"
  FAKE_DESCRIPTION = "foo_description"
  FAKE_EMAIL_PERMISSION = "Owner"
  FAKE_MEMBER_ID = "foo_member"
  FAKE_OWNER_ID = "foo_owner"

and then the use of this constants inside rspec methods? OR
if you don't like this way.. should i bring it back to the normal use like you did on User?

**Comment by davide-targa** [comment]
I am studying a way to test all the behaviors of each class.
Our current test suite is limited by the fact that we can test only some parts of the code. In particular we are not able to keep a state between tests (this is why in user_spec I used the mocking stuff). The problem is that (again) I was unable to test a considerable part of the code.
I am looking for a gem or some workaround to make tests simpler and to let us test all the behaviors.
In other words what I'm trying to achieve is a solution to your comment on issue #27...
Today I didn't go to work because of other engagements. I hope to find something useful tomorrow.
I'll let you know...

P.S.: Sorry for my english :)

**Comment by dbraga** [comment]
That's ok. i just wanted to brain storm about the way we should write our unit tests.
Let me know if you have any new info about it !

**Comment by davide-targa** [comment]
In the new specs I wrote I used faker. It's convenient, but not essential.
You're the boss, it's your choice :)
